lib/util/failover_uri.js

```diff
@@ -168,6 +168,9 @@ shaka.util.FailoverUri.prototype.createRequest_ =
         // effectively cache every response.
         this.requestPromise_ = null;
         this.request_ = null;
+        if (xhr.responseURL) {
+          this.urls.unshift(new goog.Uri(xhr.responseURL));
```

Comment:
What would be better (IMO) than prepending the URL would be to replace the specific URL this request used. (this.urls[i])